checker.c

```diff
@@ -1,21 +1,73 @@
 #include <stdio.h>
 #include <assert.h>
+#include "checker.h"
+
+
+BatteryStatus TemparatureValue = {
+        0.0,
+        45.0,
+        "Temparature"
+};
+BatteryStatus SocValue = {
+        20.0,
+        80.0,
+        "StateOfCharge"
+};
+BatteryStatus ChargeRateValue = {
+        0.0,
+        0.8,
+        "ChargeRate"
+};
+
+
+int ChecktemperatureInRange(float temperature) {
```

Comment:
This and the next two functions are very similar. Can you pass the BatteryStatus to this function, then it can be just one function?